linera-core/src/client/mod.rs

```diff
@@ -294,32 +295,51 @@ where
         validators: &[RemoteNode<impl ValidatorNode>],
         chain_id: ChainId,
         target_next_block_height: BlockHeight,
+        committee: Committee
     ) -> Result<Box<ChainInfo>, ChainClientError> {
-        // Sequentially try each validator in random order.
-        let mut validators = validators.iter().collect::<Vec<_>>();
-        validators.shuffle(&mut rand::thread_rng());
-        for remote_node in validators {
-            let info = self.local_node.chain_info(chain_id).await?;
-            if target_next_block_height <= info.next_block_height {
-                return Ok(info);
-            }
-            self.try_download_certificates_from(
-                remote_node,
-                chain_id,
-                info.next_block_height,
-                target_next_block_height,
-            )
-            .await?;
-        }
+        // Request certificates in parallel from 1/3 + 1 of total validators in terms of votes.
+        let mut validators_and_votes = validators.iter()
+            .map(|validator| (validator, committee.validators.get(&validator.public_key).expect("validator should always be in committee").votes))
+            .collect::<Vec<_>>();
+        validators_and_votes.shuffle(&mut rand::thread_rng());
+        let sufficient_threshold = (committee.total_votes() / 3) + 1;
```

Comment:
Let's use `committee.validity_threshold` here.

(It's actually not 1/3 + 1, just 1/3 rounded up.)